### ISSUE TITLE
nvim-compe feels slower than coc.nvim

### ISSUE THREAD

**Issue by HansPinckaers** [hrsh7th/nvim-compe#501]
#### Checkhealth
```
health#compe#check
========================================================================
## compe:snippet
  - INFO: snippet engine is not detected.

## compe:mapping
  - INFO: `compe#complete` is not mapped
  - INFO: `compe#confirm` is not mapped
  - INFO: `compe#close` is not mapped
  - INFO: `compe#scroll` is not mapped
```

#### Describe the bug
There seems to be a delay when filtering results. Maybe it waits for the lip servers to return completions? I think coc.nvim filters the cached completions.

I tried lowering both the throttle time and debounce text changes, but that does not seem to help.

#### To Reproduce
Type one letter after a triggerCharacter, you will see a delay in completion filtering.

#### Actual behavior
A slight delay (50ms?).

#### Expected behavior
No delay, like in coc.nvim

#### Screenshots (optional)

coc.nvim
<img src="https://user-images.githubusercontent.com/70747/127160226-3e44543e-4c6d-4c36-913b-e43481bd2645.gif" width="400">

compe
<img src="https://user-images.githubusercontent.com/70747/127159603-7fa01a5c-f93b-49ff-9891-eff31635536e.gif" width="400">

#### Additional context (optional)
Thanks a lot for all the hard work that went into this plugin!

**Comment by hrsh7th** [comment]
OK. Thank you. I will check it.

**Comment by elianiva** [comment]
have you tried setting `incomplete_delay` and `throttle_time` to 0? works fine for me

https://user-images.githubusercontent.com/51877647/127785581-bc4445ba-db79-4ad9-b645-329e6738933a.mp4

**Comment by HansPinckaers** [comment]
Hi @elianiva, yes I tried that. Sadly, the delay stays. It's more pronounced after a trigger-character (like '.'). Maybe the lsp I use is slower (jedi-language-server)? However, compe could circumvent asking the lsp for competitions when filtering down (i.e., if you have all completions of 'test.' you don't need to request them for 'test.r'), however maybe compe already does this?

**Comment by HansPinckaers** [comment]
I had some capability overwrites or pyright (to disable signature/completion), I removed those, and the delay only happens sometimes now. So it seems like there is a roundtrip to the lsp when filtering down existing completions.

**Comment by elianiva** [comment]
right, my comparison isn't that fair, I should try with the same lang server
